drivers/bluetooth/hci/siwx917_hci.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/bluetooth.h>
+
+#define DT_DRV_COMPAT silabs_siwx917_bt_hci
+#define LOG_LEVEL     CONFIG_BT_HCI_DRIVER_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_hci_driver_siwg917);
+
+#include "rsi_ble.h"
+
+static void bt_siwg917_resp_rcvd(uint16_t status, rsi_ble_event_rcp_rcvd_info_t *resp_buf);
+
+struct hci_data {
+	bt_hci_recv_t recv;
+	rsi_data_packet_t rsi_data_packet;
+};
+
+static int bt_siwg917_open(const struct device *dev, bt_hci_recv_t recv)
+{
+	struct hci_data *hci = dev->data;
+	int status = rsi_ble_enhanced_gap_extended_register_callbacks(RSI_BLE_ON_RCP_EVENT,
+								      (void *)bt_siwg917_resp_rcvd);
```

Comment:
Is the explicit `(void *)` cast really necessary?

Reply:
Yes, to avoid warning.
The function is badly designed, expecting 2 callbacks with different signature.